scipy/stats/_sobol.pyx

```diff
@@ -93,14 +88,15 @@ def initialize_direction_numbers():
         np.savez_compressed("./_sobol_direction_numbers", vinit=vs, poly=poly)
 
     """
-    cdef int[:] dns_poly
-    cdef int[:, :] dns_vinit
+    cdef cnp.uint64_t[::1] dns_poly
+    cdef cnp.uint64_t[:, :] dns_vinit
```

Comment:
I'll trust this is an appropriate data type and the change from `[:]` to `[::1]` is helpful.